charts/opentelemetry-collector/values.yaml

```diff
@@ -335,11 +335,19 @@ lifecycleHooks: {}
 #       - "5"
 
 service:
+
+  # Enable the creation of a Service pointing the agent pods with mode = daemonset.
+  # By default, Service will be created setting 'internalTrafficPolicy: Local'
+  # unless other value is explicitly set.
+  # Setting 'internalTrafficPolicy: Cluster' on a daemonset is not recommended
+  serviceForDaemonset: false
```

Comment:
Should we add `enabled` option here which will be false by default for `daemonset`, otherwise `true`?

You may ask why users would need deployment/statufulset without a service? One use case: 1-replica k8scluster receiver, which we already have a preset for. Another use case: a scraping receiver with one endpoint.

@open-telemetry/helm-approvers WDYT?

Reply:
@dmitryax how were you thinkin we set it to false by default for `mode=daemonset` only?  Are you thinkin the value is a template and `.Values.service.enabled` should always be used via a `tpl` function?

Reply:
We can do it the same way as we currently compute `internalTrafficPolicy`, right?

Reply:
I see what you mean.  Allow `enabled` as a field but don't explicitly set it.  If it is set use it, otherwise use mode to determine the default value.  Ya I like that and it keeps the section consistent with other sections in our values.yaml.  @achetronic can you make that change?

Reply:
Let's do it, probably on Monday I will upload the changes 🙏🏼

Reply:
@TylerHelmuth @dmitryax Hello there, sorry for the delay but now the approach is adapted to the discussed. Could you please review the changes? 😄